### ISSUE TITLE
Default shape and icon for favorite groups

### ISSUE THREAD

**Issue by tradavyvy** [osmandapp/OsmAnd#10850]
# 🚀 feature request

### Description

Would it be possible to choose a default shape and a default icon for each favorite group?

We use OsmAnd extensively on our motorhome trips. During the trip and in preparation beforehand we "collect" lots of favorites.
We have list/groups of "boondock" campsites; hot springs; swimming holes; slot canyons; second hand bookshops ; ......
We now use different colors for different groups but adding a default icon would make the map much more easy to read.
Our choise of light blue for swimming pool; middle blue for natural swimming hole and dark blue for hot spring is not as clear as it would be with different icons in the colored circles. And then there are national parks; state parks; arboretums; wildlife sanctuaries; .... all in 50 shades of green.

A default shape would help differentiate even more, specialy if you added a few more, like a up facing triangle (for great camping spots; swimming holes; state parks) and a down facing triangle (for the not so great ones, to remember not to visit again).

### Describe the solution you'd like

Make it be possible to choose a default shape and a default icon for each favorite group.

### Describe alternatives you've considered

Our current workaround is working with different colors, but this is way less clear and intuitive then icons.
Besides working with icons feels a lot more like the way OsmAnd uses/shows POI's.
Adding an icon manualy to each favorite is of course possible, but a lot of work and it feels unnessesary.

**Comment by vshcherb** [comment]
I think to not complicate UI or anything, we should pick up last icon / color for group that was previously selected for a previous point.

**Comment by tradavyvy** [comment]
My preferente would be to revert to a fixed default color/icon every time nothing specific was picked manualy.
I see 2 kinds of users:
---Specific category users like me;
As example group hotsprings, almost always I want to use the spring icon on dark blue, but sporadicly I like to use the stream icon on dark blue for a hot-river site, but the next one is more than likely a normal hotspring again so I would like to revert to the default.
--- Generic category users;
As example group roadside atractions, usualy the picked icons wil be mixed -museum -amusement park -zoo -another museum -viewpoint -viewpoint -pool -viewpoint -park ..... With probably one frequent icon mixed in (viewpoint, miscelaneous, uncategorised ...)
Both user categories would be best served with a default color/icon choice.

But last user color/icon would be a big improvement already.

**Comment by tradavyvy** [comment]
Also consider osmands behaviour when you want to change the color/icon/shape for a whole group.

In case of "previous selected" it would be hard to know which favorites in the group to change and which not.
leaving 2 undesirable options: 
---change everything and loose all previous information
---change nothing and make groupwise changing impossible

In case of "fixed defaults" Osmand could change only the favorites with the "old" defaults into the "new" default and thus keep all the previous existing information.

**Comment by Boothy99** [comment]
Good to see the issue here and as with #9104 is still open and on the books.

**Comment by Friptick** [comment]
Firstly, @vshcherb, why not close #9104? They are fundamentally the same thing and we are discussing details in two places. That issue is shelved as "Nice to have" while this one is milestoned.

My attempt to sum up the problem: **Currently it is not possible to visually differentiate more than about 9 groups.** There are 12 colors to choose from, but "orange" and "red" are almost indistinguishable, as are "dark yellow" and "yellow", as are "purple" and "pink".

This may seem like a small problem but I too have been trying to work around it for years, like @tradavyvy

So, two obvious ways to solve the problem:

- Allow icons for groups in addition to colors, as @tradavyvy proposes (but maybe it is then necessary to add a setting for icon size - personally I do not use icons at all because I cannot  easily distinguish them except by the hack of increasing the text size)
- Add colors, or make them freely definable via a color-picker

Personally I do not mind how the issue is solved. But the fundamental issue is that there is currently a functional limit on tthe number of groups, and it needs to be raised.

Keep up the good work.

**Comment by tradavyvy** [comment]
Bummer
I just noticed @vshcherb moved the milestone to 4.2.
Ofcourse I can't see into the devellopers workload, nor do I know how difficult this is to implement but PLEASE give this some priority.
As I said functions like this set osmand appart from the competition. On things like userbase and traffic info and even navigation you are probably never going you beat the big G...
But on specialised things like offline use (G is not interested becourse they don't get your data) and personal map-layout (G caters only to the masses) you are and could stay well ahead of the curve.

**Comment by Friptick** [comment]
>On things like userbase and traffic info and even navigation you are probably never going you beat the big G... But on specialised things like offline use (G is not interested becourse they don't get your data) and personal map-layout (G caters only to the masses) you are and could stay well ahead of the curve. (@tradavyvy)

Yes I tend to agree, but one can sympathize with the devs too.

First, they and @vshcherb are doing all this for free. So effectively we are just begging for crumbs, whether we like that or not. Even if the app maintainers are misguided we still have no right to complain. It is just not a commercial relationship. 

Second, hardly anybody is asking for this feature! Doesn't mean it is not a good idea, but presumably if any of the devs needed this themselves, it would have been fixed. We are asking them to work for free and almost nobody is speaking up to support us. Obviously many users don't even know what an issue tracker is, so perhaps a few are suffering in silence. But you see the problem.

All that said, I absolutely agree with your point. **Osmand cannot beat Google at Google's own game, but it absolutely can (already does) beat it at anything that is not commercial.** Offline, privacy, and anything which leverages OSM's killer feature: crowd-sourced data about the outdoors and generally places where there are no shops.

**Comment by vshcherb** [comment]
Indeed we have already a nice design / solution for both of the problems though they won't be a simple code change as I've thought in the beginning. This user story became quite large we hope to finish it this year in 4.2 or 4.3 release

**Comment by dmpr0** [comment]
Done, available in the latest nightly builds.

**Comment by Boothy99** [comment]
Many thanks for adding the new ability to completely edit POI appearances !! Excellent work and much appreciated.

**Comment by pebogufi** [comment]
@vshcherb
Nearly everything works fine now.
But if i move a favourite to a different group and want to change the color of item to that of this "to-group" (so that all colors of group are identically), this is no more possible, if the to-group has a user defined color.
- Please add in edit dialog the color of to-group in list of colors also if color is user defined.
- and please increase the possible number of user defimed colors

**Comment by tradavyvy** [comment]
See also #14245

(Great improvements uptill now though, THANKS)

**Comment by pebogufi** [comment]
As a workaround i can reset a color of a single favourite to group default:
Use a user defined color, set it to 00000000, then set it to this favourite and save.

**Comment by Friptick** [comment]
Please re-open. As per @pebogufi above, this feature only half works, and creates unexpected results.

To reproduce:
1. "Add favorite" and give it a new group "Restaurants" with default color "red"
2. "Add favorite" again, but this time with group "Theaters" which has default color "blue"
3. Now we have two favorites, in two groups, with two colors - all as expected
4. "Edit favorite" for the first one > change its group from "Restaurants" to "Theaters"
5. **Boom! Now there are two favorites, both in the same group, but with two different colors**

In other words: groups have colors, but moving  a favorite from one group to another does not change its color. This is confusing and unexpected behavior. And of course it makes it impossible to change the appearance of a group's POIs all at once, which is surely half the point of having groups.  

For this feature to be finished, the appearance attributes really need to be attached to the groups, not to the individual POIs.

Thanks.

**Comment by pebogufi** [comment]
> As a workaround i can reset a color of a single favourite to group default:
Use a user defined color, set it to 00000000, then set it to this favourite and save.

This does not work anymore - it's a pity...

**Comment by Friptick** [comment]
Yep. Static POIs do not seem to be a priority for Osmand. The focus seems to be all on navigation. Fair enough, navigation functionality is a hard technical challenge compared to favorites, and lots of users seem to want it.

But it would be great to have an OSM client geared to static POIs. I'm keeping an eye on Organic Maps.

**Comment by vshcherb** [comment]
You can still change appearance to the group and apply it. It works as expected cause probably some users want to preserve original colors of the favorite.

**Comment by vshcherb** [comment]
You can lock appearance to the group that was not an intention, we have functionality of Default appearance but it doesn't mean Group Constant Appearance which is a different use case

**Comment by Friptick** [comment]
>It works as expected cause probably some users want to preserve original colors of the favorite.

But it is possible to change a single favorite's color without using groups at all. Conversely, if the user creates a group with certain attributes, it seems quite likely they expect them to be attributes of the group, no? I certainly did. I assumed it was just a bug when I moved a favorite from one group to another and found the color "orphaned". I mean, what is a more obvious use for a group than to control the visual appearance of the points _as a group_ and not individually?

>Group Constant Appearance which is a different use case

And surely the most common one.

It also has another major advantage: portability. Consider this concrete example that I have been struggling with (to the point of writing a Python script to fix it):

1. Say I have two groups, "Parks" and "Pools". The parks should be green and the pools should be blue. Logical.
2. In `track.gpx` or `favorites.gpx` the markup is `<type>Park</type>` and `<type>Pool</type>`. Great! Now I can add a new park to the file using my desktop mapping software. I will just give it the type "Park", because it styles Parks as green.
3. Problem! Osmand's color information is attached _individually_ to the points in the extensions namespace: `<osmand:color>#000001</osmand:color>` etc. So I'm out of luck. `<type>Park</type>` can be green in my desktop app and green in Osmand, but the POI will only be green in Osmand if it was created in Osmand. 

TLDR: In Osmand, "group" equates to GPX's "type". To function properly, "type" need to be stylable. Please re-open or confirm this can be raised as another issue.

**Comment by vshcherb** [comment]
I don't agree that request was in scope of original implementation and there was no intention to support it in that way. Anyway Reopen is not possible cause it was fixed already in 4.2. 

You can open new issue in following context that you would like apply appearance of newly created items (already implemented) and moved items to the group (not implemented yet). 

We think to have either setting or a dialog that will keep asking to keep or change, so it something to think.
